src/js/login/components/SearchHelpSignIn.jsx

```diff
@@ -1,103 +1,100 @@
+import PropTypes from 'prop-types';
 import React from 'react';
-import { connect } from 'react-redux';
 import _ from 'lodash';
-import URLSearchParams from 'url-search-params';
 import classNames from 'classnames';
 
 import recordEvent from '../../../platform/monitoring/record-event';
 import HelpMenu from '../../common/components/HelpMenu';
 import SearchMenu from '../../common/components/SearchMenu';
 import SignInProfileMenu from './SignInProfileMenu';
 
-import { toggleLoginModal, toggleSearchHelpUserMenu } from '../actions';
-import { isUserRegisteredForBeta } from '../../personalization/beta-enrollment/actions';
-
-export class SearchHelpSignIn extends React.Component {
-  componentDidUpdate(prevProps) {
-    const { currentlyLoggedIn, showModal } = this.props.login;
-    const isModalStillClosed = !prevProps.login.showModal && !showModal;
-    if (!currentlyLoggedIn && isModalStillClosed) {
-      const nextParams = new URLSearchParams(window.location.search);
-      const nextPath = nextParams.get('next');
-      if (nextPath) {
-        this.props.toggleLoginModal(true);
-      }
-    }
-  }
-
+class SearchHelpSignIn extends React.Component {
   handleSignInSignUp = (e) => {
     e.preventDefault();
     recordEvent({ event: 'login-link-clicked' });
     this.props.toggleLoginModal(true);
   }
 
-  hasSession() {
+  handleMenuClick = (menu) => (() => {
```

Comment:
This is somewhat hard to read for me. I'm not entirely sure we need the auto-binding part of this (I think that's happening on lines 22-24) and could maybe do something like this:

```js
handleMenuClick(menu) => {
  return () => this.props.toggleMenu(menu, !this.props.isMenuOpen[menu]);
}
```

Not sure, though.

Reply:
I was going for a more functional style, like
```
someFunction = (arg) => (anotherFunction);
```
, but it is a little harder to read without the `return`, so I can rewrite it to be more readable.

I did the binding on the following lines because I didn't want to recreate those functions on each render cycle (like when a menu is toggled).